src/FunctionApps/python/tests/IntakePipeline/conversion_test.py

```diff
@@ -0,0 +1,171 @@
+from unittest import mock
+import pytest
+
+from IntakePipeline.conversion import (
+    clean_message,
+    convert_batch_messages_to_list,
+    convert_message_to_fhir,
+    get_file_type_mappings,
+)
+
+
+def test_clean_message():
+    TEST_STRING1 = "\r\nHello\nWorld\r\n"
+    TEST_STRING2 = "\nHello\r\nW\r\norld"
+    TEST_STRING3 = "Hello World"
+    TEST_STRING4 = "\u000bHello World\u001c"
+    TEST_STRING5 = "\u000bHello\r\nWorld\u001c"
+
+    assert clean_message(TEST_STRING1) == "Hello\nWorld"
+    assert clean_message(TEST_STRING2) == "Hello\nW\norld"
+    assert clean_message(TEST_STRING3) == "Hello World"
+    assert clean_message(TEST_STRING4) == "Hello World"
+    assert clean_message(TEST_STRING5) == "Hello\nWorld"
+
+
+def test_convert_batch_messages_to_list():
+    TEST_STRING1 = """
+    MSH|blah|foo|test
+    PID|some^text|blah
+    OBX|foo||||bar^baz&foobar
+    MSH|blah|foo|test
+    PID|some^text|blah
+    OBX|foo||||bar^baz&foobar
+    """.replace(
+        " ", ""
+    )
+
+    TEST_STRING2 = """
+    FHS|^~&|WIR11.3.2|WIR|||20200514||1219144.update|||
+    BHS|^~&|WIR11.3.2|WIR|||20200514|||||
+    BTS|0|
+    FTS|1|
+    """.replace(
+        " ", ""
+    )
+
+    TEST_STRING3 = """
+    FHS|^~&|WIR11.3.2|WIR|||20200514||1219144.update|||
+    BHS|^~&|WIR11.3.2|WIR|||20200514|||||
+    MSH|^~&|WIR11.3.2^^|WIR^^||WIRPH^^|20200514||VXU^V04|2020051411020600|P^|2.4^^|||ER
+    PID|||3054790^^^^SR^~^^^^PI^||ZTEST^PEDIARIX^^^^^^|HEPB^DTAP^^^^^^|20180808|M|||||||||||||||||||||
+    PD1|||||||||||02^^^^^|Y||||A
+    NK1|1||BRO^BROTHER^HL70063^^^^^|^^NEW GLARUS^WI^^^^^^^|
+    BTS|5|
+    FTS|1|
+    """.replace(
+        " ", ""
+    )
+
+    list1 = convert_batch_messages_to_list(TEST_STRING1)
+    list2 = convert_batch_messages_to_list(TEST_STRING2)
+    list3 = convert_batch_messages_to_list(TEST_STRING3)
+
+    assert len(list1) == 2
+    assert len(list2) == 0
+    assert len(list3) == 1
+
+    assert list1[0].startswith("MSH|")
+    assert list1[1].startswith("MSH|")
+    assert list3[0].startswith("MSH|")
+
```

Comment:
You could _potentially_ add another set of assertions for the entries in list 1 having all three group strings in them (i.e. MSH, PID, OBX each), but that's probably overkill and these tests would satisfy me as is.